catalog/core/catalog-core-directorymonitor/src/main/java/org/codice/ddf/catalog/content/monitor/DurableWebDavFileConsumer.java

```diff
@@ -0,0 +1,134 @@
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.ddf.catalog.content.monitor;
+
+import java.io.IOException;
+import java.nio.file.Path;
+import java.nio.file.StandardWatchEventKinds;
+import java.nio.file.WatchEvent;
+import java.util.List;
+
+import org.apache.camel.Exchange;
+import org.apache.camel.Processor;
+import org.apache.camel.component.file.GenericFileEndpoint;
+import org.apache.camel.component.file.GenericFileOperations;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.github.sardine.Sardine;
+import com.github.sardine.SardineFactory;
+
+public class DurableWebDavFileConsumer extends AbstractDurableFileConsumer {
+    private static final Logger LOGGER =
+            LoggerFactory.getLogger(DurableFileSystemFileConsumer.class);
+
+    private DavAlterationObserver observer;
+
+    private EntryAlterationListener listener = new EntryAlterationListenerImpl();
+
+    private Sardine sardine = SardineFactory.begin();
+
+    DurableWebDavFileConsumer(GenericFileEndpoint<EventfulFileWrapper> endpoint,
+            Processor processor, GenericFileOperations<EventfulFileWrapper> operations) {
+        super(endpoint, processor, operations);
+    }
+
+    @Override
+    protected boolean pollDirectory(String fileName, List list, int depth) {
+        String remaining = ((DurableFileComponent) endpoint.getComponent()).remaining;
+        initialize(remaining);
+        if (observer != null) {
+            observer.addListener(listener);
+            observer.checkAndNotify(sardine);
+            observer.removeListener(listener);
+            fileSystemPersistenceProvider.store(String.valueOf(remaining.hashCode()), observer);
+            return true;
+        } else {
+            return isMatched(null, null, null);
+        }
+    }
+
+    private void initialize(String fileName) {
+        if (fileSystemPersistenceProvider == null) {
+            fileSystemPersistenceProvider =
+                    new FileSystemPersistenceProvider(getClass().getSimpleName());
+        }
+        if (observer == null && fileName != null) {
+            if (fileSystemPersistenceProvider.loadAllKeys()
+                    .contains(String.valueOf(fileName.hashCode()))) {
+                observer =
+                        (DavAlterationObserver) fileSystemPersistenceProvider.loadFromPersistence(
+                                String.valueOf(fileName.hashCode()));
+            } else {
+                observer = new DavAlterationObserver(new DavEntry(fileName));
+            }
+        }
+    }
+
+    @Override
+    public void shutdown() throws Exception {
+        super.shutdown();
+        if (observer != null) {
+            observer.destroy();
+        }
+    }
+
+    private void createExchangeHelper(DavEntry entry, WatchEvent.Kind<Path> fileEvent) {
```

Comment:
❓ since this is a private method and only called in the `EnteryAlterationListnerImpl` class, should it just be moved there or is there a bigger purpose to this method ?